expan/core/experiment.py

```diff
@@ -190,36 +190,51 @@ def filter(self, kpis, percentile=99.0, threshold_type='upper'):
 
         self.data = self.data[flags == False]
 
-    def sga(self, feature_name_to_bins):
+    def sga(self, feature_name_to_bins, time_interval=None):
         """
-        Perform subgroup analysis.
-    
+        Perform subgroup analysis. If start and end dates are specified, the time-based sga is performed alongside.
+
         Args:
             feature_name_to_bins (dict): a dict of feature name (key) to list of Bin objects (value). 
                                       This dict defines how and on which column to perform the subgroup split.
-                                      
+            time_interval: lower and upper bound of the time interval for the date of treatment.
+
         Returns:
             Analysis results per subgroup. 
         """
+
         for feature in feature_name_to_bins:
             # check type
             if type(feature) is not str:
-                raise TypeError("Key of the input dict needs to be string, indicating the name of dimension")
+                raise TypeError("Key of the input dict needs to be string, indicating the name of dimension.")
             if type(feature_name_to_bins[feature]) is not list:
                 raise TypeError("Value of the input dict needs to be a list of Bin objects.")
             # check whether data contains this column
             if feature not in self.data:
                 raise KeyError('No column %s provided in data.' % feature)
 
         subgroups = []
-        for feature in feature_name_to_bins:
-            for bin in feature_name_to_bins[feature]:
-                subgroup = {'dimension': feature,
-                            'segment': str(bin.representation)}
-                subgroup_data = bin.apply(self.data, feature)
-                subgroup_res = self._delta(method='fixed_horizon', data=subgroup_data,
-                                           num_tests=len(self.report_kpi_names))
-                subgroup['result'] = subgroup_res
-                subgroups.append(subgroup)
-
+        if feature_name_to_bins:
+            for feature in feature_name_to_bins:
+                for bin in feature_name_to_bins[feature]:
+                    subgroup = {'dimension': feature,
+                                'segment': str(bin.representation)}
+                    subgroup_data = bin.apply(self.data, feature)
+                    subgroup_res = self._delta(method='fixed_horizon', data=subgroup_data,
+                                               num_tests=len(self.report_kpi_names))
+                    subgroup['result'] = subgroup_res
+                    subgroups.append(subgroup)
+
+        if time_interval:
```

Comment:
Here again - wouldn't it be slightly more general not to try to make a special case of binning in time?